backend/src/main/java/com/htecgroup/skynest/service/impl/BucketServiceImpl.java

```diff
@@ -103,10 +104,21 @@ public void deactivateLambda(UUID bucketId, LambdaType lambda) {
 
   @Override
   public List<BucketResponse> listAllBuckets() {
+
     List<BucketEntity> entityList =
         (List<BucketEntity>) bucketRepository.findAllByOrderByNameAscCreatedOnDesc();
+    List<BucketEntity> newEntityList = new ArrayList<BucketEntity>();
+
+    for (BucketEntity bucket : entityList) {
```

Comment:
Use streams for this, filter method is perfect

Reply:
the filter predicate could be something like this

```
try {
  permissionService.currentUserHasPermissionForBucket(bucket.getId(), AccessType.VIEW);
  newEntityList.add(bucket);
} catch (BucketAccessDeniedException b) {
  return false;
}
return true;
```